client/ayon_deadline/plugins/publish/global/collect_deadline_job_env_vars.py

```diff
@@ -53,3 +51,17 @@ def process(self, context):
 
         # Transfer some environment variables from current context
         context.data.setdefault(JOB_ENV_DATA_KEY, {}).update(env)
+
+
+class CollectAYONServerUrlToFarmJob(CollectDeadlineJobEnvVars):
+    label = "Submit AYON server URL to farm job"
+    settings_category = "deadline"
+
+    # Enable in settings
+    enabled = False
+
+    ENV_KEYS = [
+        "AYON_SERVER_URL",
+        "AYON_API_KEY",
```

Comment:
I don't like to pass api key to the job, this is huge blocker.

Reply:
Valid point - the implementation here is mostly for DEV servers. I would like it, and @tweak-wtf too. We could of course mark it as such and put a description on settings that it's intended exactly for that?

And.. a better improvement would be to define in AYON settings for this plug-in which API Key to send along? So that you can still set it to the service user API key. Would that solve your concerns?

Reply:
`AYON_API_KEY` is not artist api key, it should be service API key for connecting DL to AYON.  
It could be there, if one would need providing API key value different for each Deadline. In that case value currently set in Deadline AYON plugin wouldnt be enough.
![image](https://github.com/user-attachments/assets/5768c66f-f5d1-441c-8430-fdcf6e030ed3)

Reply:
> `AYON_API_KEY` is not artist api key, it should be service API key for connecting DL to AYON. It could be there, if one would need providing API key value different for each Deadline. In that case value currently set in Deadline AYON plugin wouldnt be enough.

I suppose you're replying to @iLLiCiTiT 's comment. So you agree with having `AYON_API_KEY` to be specified in AYON studio/project settings for this particular addon would be a good workaround @kalisp ?

Reply:
I was testing it by adding `AYON_API_KEY` to `ayon+settings://core/project_environments?project=test_project_1` and it was working. 

I think this is enough for such a edge case. 

I just dont like that `Submit ...` label.

I feel neither here nor there about Kuba comment to smush it into single collector `CollectDeadlineJobEnvVars`

![image](https://github.com/user-attachments/assets/b02842d0-a30e-49c1-b39b-a193ab2aedb6)

![image](https://github.com/user-attachments/assets/99415fc3-b820-41d6-b889-7dc7621d5d9b)

Reply:
> I was testing it by adding `AYON_API_KEY` to `ayon+settings://core/project_environments?project=test_project_1` and it was working.
> (Without it it was picking value from DL AYON plugin correctly, was worried empty value would override it.)

This couldn't actually work on the `GlobalJobPreLoad` because it needs the API key to connect to the server first - so it can't access the settings to get that API key in the first place. :)

Reply:
What? In `GlobalJobPreLoad` it now checks if job has that env var, if it does, it uses it, if not, it looks for api key from DL plugin.

And that is how it is actually supposed to work. (Or I misunderstood.)
![image](https://github.com/user-attachments/assets/a5339f5f-89f7-4eeb-ac75-40e144af1dfa)

Reply:
This adds AYON_API_KEY to the job during publishing. Which means it adds artist api key.

Reply:
> I was testing it by adding AYON_API_KEY to ayon+settings://core/project_environments?project=test_project_1 and it was working.

How do you access project settings without api key? You must have artist api key passed to job to be able to do it.

Reply:
Yes, I am stupid, I was sure that when I tested it for the first time, logging of AYON_API_KEY was empty in that collector, so I thought that env var is unoccupied in regular AYON, until its is provided by global environment.

It seems its not the case, it holds actually ARTIST key by default.

But new collector is not replacing it, just reusing it, if there is no value in Settings.

Note about providing API key is in hover message and it will be in documentation, so I think it should be fine.